src/App.tsx

```diff
@@ -1,12 +1,63 @@
 import React from 'react';
 import './App.scss';
 
-// import { getAll, get5First, getRed } from './api/goods';
-// or
-// import * as goodsAPI from './api/goods';
+import { Goodslist } from './components/GoodsList/GoodsList';
+import * as goodsAPI from './api/goods';
 
-const App: React.FC = () => (
-  <h1>Dynamic list of Goods</h1>
-);
+type State = {
+  goods: Good[];
+};
+
+class App extends React.PureComponent<{}, State> {
+  state: State = {
+    goods: [],
+  };
+
+  displayGoods = async (selected : () => Promise<Good[]>) => {
+    const goods = await selected();
+
+    this.setState({ goods });
+  };
+
+  render() {
+    const { goods } = this.state;
+
+    return (
+      <div className="app">
+        <h1 className="app__title">Dynamic list of Goods</h1>
+        <div className="app__buttons">
+          <button
+            type="button"
+            className="app__button"
+            onClick={() => (
+              this.displayGoods(goodsAPI.getAll)
+            )}
+          >
+            Load All goods
+          </button>
+          <button
+            type="button"
+            className="app__button"
+            onClick={() => (
+              this.displayGoods(goodsAPI.get5First)
+            )}
+          >
+            Load 5 first goods
+          </button>
+          <button
+            type="button"
+            className="app__button"
+            onClick={() => (
+              this.displayGoods(goodsAPI.getRedGoods)
+            )}
+          >
+            Load red goods
+          </button>
+        </div>
+        <Goodslist goods={goods} />
```

Comment:
What will the user see while goods are loading?

Reply:
You're right, fixed.